### DISCUSSION TITLE
[IDEA] Restore Backup with Merge to existing data option?

### DISCUSSION THREAD

**Discussion by johnkennyuk** [pieces-app/support#427] [Ideas]
I do like the Backup & Restore feature in Pieces, but I have run into problems before whereby my appdata drops out for whatever reason and I haven't immediately noticed, when I do notice and then have to restore from a backup I currently lose all the existing data I've recently created.  

My Idea is to offer the option to be able to either $${\color{red}Merge \space the \space restore \space point \space to \space the \space current \space data}$$ as well as the option to what we see now which is to overwrite the current data..

Thanks

**Comment by rosie-at-pieces** [comment]
@mark-at-pieces @hiro-at-pieces good feedback for our full backup and sync functionality!

**Comment by rosie-at-pieces** [comment]
[#1297](https://github.com/pieces-app/os_server/issues/1297)

**Comment by mark-at-pieces** [comment]
As of right now you are absolutely correct this will just enable you you fully restore a previous snapshot of your database, which will over ride any changes that were not backed up to the cloud.

That being said this is top of mind and in development at the moment, we hope to get this out along with improved stability to all existing cloud features, hopefully 🤞 before Christmas for a nice little gift for each of our Pieces users!
